src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -234,17 +234,27 @@ fn search(
     total_space: Sectors,
     upper_limit: DataBlocks,
     lower_limit: DataBlocks,
+    current_meta_size: Sectors,
     fs_limit: u64,
 ) -> StratisResult<(Sectors, Sectors)> {
     let diff = upper_limit - lower_limit;
     let half = lower_limit + diff / 2u64 + diff % 2u64;
-    let half_meta = thin_metadata_size(DATA_BLOCK_SIZE, datablocks_to_sectors(half), fs_limit)?;
+    let half_meta = max(
+        thin_metadata_size(DATA_BLOCK_SIZE, datablocks_to_sectors(half), fs_limit)?,
+        current_meta_size,
+    );
 
     if diff == DataBlocks(1) {
```

Comment:
suggest additional distinct termination condition in case ```current_data_size``` > than optimum data size.

```
if half == current_data_size {
    Ok((current_data_size, (total - current_data_size) / 2))
}
```

Reply:
> suggest additional distinct termination condition in case `current_data_size` > than optimum data size.
> 
> ```
> if half == current_data_size {
>     Ok((current_data_size, (total - current_data_size) / 2))
> }
> ```
Thinking you should return an error here, because metadata size can not be enough.

Reply:
But also thinking that an error is wrong, because in the usual overprovisioned case, even if the metadata is not enough, then you still want to expand. Otoh, in the switch to no-overprovisioned calculation, I think you want to reject the result, because there isn't room enough for the metadata. So, maybe it would be best to enrich the return type somehow.